File: app/src/main/java/com/example/argohunt/models/HuntInfo.java
package com.example.argohunt.models;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class HuntInfo
{
    private static HuntInfo sHunt;
    private List<ArgoHunt> mHunts;

    public static HuntInfo get(Context context)
    {
        if(sHunt == null)
        {
            sHunt = new HuntInfo(context);
        }

        return sHunt;
    }

    private HuntInfo(Context context)
    {
        mHunts = new ArrayList<>();

    }

    public List<ArgoHunt> getmHunts()
    {
        return mHunts;
    }

    public ArgoHunt getHunt(UUID id)
    {
        for(ArgoHunt Hunt: mHunts)
        {
            if(Hunt.getmId().equals(id))
            {
                return Hunt;
            }
        }

        return null;
    }
}

File: app/src/main/java/com/example/argohunt/models/HuntInfo.kt
package com.example.argohunt.models

import android.content.Context
import java.util.*

class HuntInfo private constructor(context: Context)
{
    private val mHunts: List<ArgoHunt>

    fun getmHunts(): List<ArgoHunt>
    {
        return mHunts
    }

    fun getHunt(id: UUID): ArgoHunt?
    {
        for (Hunt in mHunts)
        {
            if (Hunt.getmId() == id)
            {
                return Hunt
            }
        }
        return null
    }

    companion object
    {
        private var sHunt: HuntInfo? = null
        operator fun get(context: Context): HuntInfo?
        {
            if (sHunt == null)
            {
                sHunt = HuntInfo(context)
            }
            return sHunt
        }
    }

    init
    {
        mHunts = ArrayList()
    }
}